pingcap/tidb
# *: add an option to enable pessimistic mode for auto-commit transactions

Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>



### What problem does this PR solve?


Issue Number: close #31135

Problem Summary:
Add an option to enable pessimistic for auto-commit transactions.

### What is changed and how it works?

### Check List

Tests 

1. start a cluster and set `pessimistic-auto-commit = true` in config file
2. run sysbench update non index on pessimistic mode
3. check the results
<img width="602" alt="截屏2022-01-30 下午3 51 30" src="https://user-images.githubusercontent.com/4352397/151691402-6d53a63a-9359-497c-8138-dea6dbd7b1f6.png">


Side effects


Documentation


### Release note



```release-note
Add an option to enable pessimistic mode for auto-commit transactions.
```
